### PR TITLE
tailcfg, control/controlclient: add tailcfg.PeersChangedPatch [capver 33]

### PR DESCRIPTION
**Pull request by bradfitz**
This adds a lighter mechanism for endpoint updates from control.

### CODE REVIEW

File: control/controlclient/map.go [hunk 1]
```diff
@@ -259,6 +260,16 @@ func undeltaPeers(mapRes *tailcfg.MapResponse, prev []*tailcfg.Node) {
 				n.Online = &online
 			}
 		}
+		if ec := mapRes.EndpointChange; ec != nil {
+			if n, ok := peerByID[ec.ID]; ok {
+				if ec.DERPRegion != 0 {
```
Comment (maisem) [inline control/controlclient/map.go:265]: no way to say empty?

**Review by maisem — APPROVED** [review]

File: control/controlclient/map.go (hunk 1 above)
Comment (bradfitz) [reply control/controlclient/map.go:265]: kinda pointless. we have Offline already

**Comment by bradfitz** [comment]
PTAL; changed to slice & renamed type/field

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -67,7 +67,8 @@ type CapabilityVersion int
 //    30: 2022-03-22: client can request id tokens.
 //    31: 2022-04-15: PingRequest & PingResponse TSMP & disco support
 //    32: 2022-04-17: client knows FilterRule.CapMatch
-const CurrentCapabilityVersion CapabilityVersion = 32
+//    33: 2022-07-20: client knows MapResponse.EndpointChange
```
Comment (maisem) [inline tailcfg/tailcfg.go:70]: no longer EndpointChange

Comment (bradfitz) [reply tailcfg/tailcfg.go]: fixed, and commit message

File: tailcfg/tailcfg.go [hunk 3]
```diff
@@ -67,7 +67,8 @@ type CapabilityVersion int
 //    30: 2022-03-22: client can request id tokens.
 //    31: 2022-04-15: PingRequest & PingResponse TSMP & disco support
 //    32: 2022-04-17: client knows FilterRule.CapMatch
-const CurrentCapabilityVersion CapabilityVersion = 32
+//    33: 2022-07-20: client knows MapResponse.PeersChangedPatch
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply tailcfg/tailcfg.go]: done

**Review by maisem — APPROVED** [review]

**Pull request merged**